scylla/src/transport/iterator.rs

```diff
@@ -364,6 +364,10 @@ where
                         continue 'nodes_in_plan;
                     }
                     RetryDecision::DontRetry => break 'nodes_in_plan,
+                    RetryDecision::IgnoreWriteError => {
+                        warn!("Ignoring error during fetching pages; stopping fetching.");
+                        return;
+                    }
```

Comment:
I'm not sure at all, but I have no idea what to do instead. Any suggestions?

Reply:
Now that I think about it, there are two potentially problematic cases:

1. A user implements a buggy retry policy which returns `IgnoreWriteError` on failed page fetch,
2. A user unnecessarily uses an `_iter` interface to perform a write operation and it returns `IgnoreWriteError`.

In case 1, it makes more sense to return an error - on the other hand, in case of 2 it makes more sense to silently return. I don't think it is possible to differentiate, so I'm not sure which would be better. @cvybhu  @psarna WDYT?

Reply:
Let's not optimize for buggy retry policies, so I suggest we stop considering 1. at and just return an error

Reply:
Just to make sure: we settle on case 2., which means returning with no error (but possibly a warning)?

Reply:
ugh, my response was a contradictory brain fart... yes, let's focus on handling case 2, so continuing with emitting a warning